src/interpreter/html.rs

```diff
@@ -248,3 +252,120 @@ pub enum Element {
     Details(Section),
     Summary,
 }
+
+impl Element {
+    pub fn table() -> Self {
+        Self::Table(Table::new())
+    }
+
+    pub fn table_row() -> Self {
+        Self::TableRow(Vec::new())
+    }
+
+    pub fn unordered_list() -> Self {
+        Self::List(List {
+            ty: ListType::Unordered,
+        })
+    }
+
+    pub fn ordered_list(start_index: usize) -> Self {
+        Self::List(List {
+            ty: ListType::Ordered(start_index),
+        })
+    }
+
+    pub fn as_mut_list(&mut self) -> Option<&mut List> {
+        if let Self::List(list) = self {
+            Some(list)
+        } else {
+            None
+        }
+    }
+
+    pub fn as_mut_table(&mut self) -> Option<&mut Table> {
+        if let Self::Table(table) = self {
+            Some(table)
+        } else {
+            None
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum TagName {
+    Anchor,
+    BlockQuote,
+    BoldOrStrong,
+    Break,
+    Code,
+    Details,
+    Div,
+    EmphasisOrItalic,
+    Header(HeaderType),
+    HorizontalRuler,
+    Image,
+    Input,
+    ListItem,
+    OrderedList,
+    Paragraph,
+    PreformattedText,
+    Section,
+    Small,
+    Span,
+    Strikethrough,
+    Summary,
+    Table,
+    TableBody,
+    TableDataCell,
+    TableHead,
+    TableHeader,
+    TableRow,
+    Underline,
+    UnorderedList,
+}
```

Comment:
This is an important and large part of the interpreter, I'd suggest moving it to another subfile.
`Element` could also benefit from moving to another subfile